src/main/java/SYSC4806/Controller/HomeController.java

```diff
@@ -101,8 +103,37 @@ public String showResults(Model model, @PathVariable("program") String p, @PathV
         return "fragments/results :: resultsTable";
     }
 
+
+    @GetMapping("/add/{name}/{code}/{year}/{category}/{learningO}/{program}")
+    public String add(Model model, @PathVariable("name") String n, @PathVariable("code") String co, @PathVariable("year")
+            String y, @PathVariable("category") String ca, @PathVariable("learningO") String lo,
+                      @PathVariable("program") String p)
+    {
+        Category cat = new Category(ca);
+        categoryRepository.save(cat);
+
+        Course course =  new Course(Integer.parseInt(y), n, co);
+        courseRepository.save(course);
+
+        LearningOutcome outcome = new LearningOutcome(lo, cat);
+        outcome.addCourse(course);
+        learningOutcomeRepository.save(outcome);
+
+        Program pro = new Program(p);
+        pro.addCourse(course);
+        programRepository.save(pro);
+
```

Comment:
would this cause any issues if the values for the variables are null or empty? If the admin didn't add all the information I guess that could be a issue so maybe have some checks for it